app/scripts/worker.js

```diff
@@ -106,47 +108,70 @@ export function workerSetPix(
       .domain(valueScaleDomain);
   }
 
-  const pixData = new Uint8ClampedArray(size * 4);
+  let filteredSize = size;
+  if (shape && selectedRows) {
+    filteredSize = selectedRows.length * shape[1];
+  }
 
-  let rgbIdx = 0;
-  let d;
   const tileWidth = Math.sqrt(size);
+  const pixData = new Uint8ClampedArray(filteredSize * 4);
+  let rgbIdx;
+
+  const setPixData = (i, d) => {
+    // Transparent
+    rgbIdx = 255;
+
+    if (
+      // ignore the upper right portion of a tile because it's on the diagonal
+      // and its mirror will fill in that space
+      !(ignoreUpperRight && Math.floor(i / tileWidth) < i % tileWidth) &&
+      !(ignoreLowerLeft && Math.floor(i / tileWidth) > i % tileWidth) &&
+      // Ignore color if the value is invalid
+      !Number.isNaN(+d)
+    ) {
+      // values less than espilon are considered NaNs and made transparent (rgbIdx 255)
+      rgbIdx = Math.max(
+        0,
+        Math.min(254, Math.floor(valueScale(d + pseudocount)))
+      );
+    }
+    // let rgbIdx = qScale(d); //Math.max(0, Math.min(255, Math.floor(valueScale(ct))))
+    if (rgbIdx < 0 || rgbIdx > 255) {
+      console.warn(
+        'out of bounds rgbIdx:',
+        rgbIdx,
+        ' (should be 0 <= rgbIdx <= 255)'
+      );
+    }
+    const rgb = colorScale[rgbIdx];
+
+    pixData[i * 4] = rgb[0];
+    pixData[i * 4 + 1] = rgb[1];
+    pixData[i * 4 + 2] = rgb[2];
+    pixData[i * 4 + 3] = rgb[3];
+  };
 
+  let d;
   try {
-    for (let i = 0; i < data.length; i++) {
-      d = data[i];
-
-      // Transparent
-      rgbIdx = 255;
-
-      if (
-        // ignore the upper right portion of a tile because it's on the diagonal
-        // and its mirror will fill in that space
-        !(ignoreUpperRight && Math.floor(i / tileWidth) < i % tileWidth) &&
-        !(ignoreLowerLeft && Math.floor(i / tileWidth) > i % tileWidth) &&
-        // Ignore color if the value is invalid
-        !Number.isNaN(+d)
+    if (filteredSize !== size) {
```

Comment:
Would you mind adding some comments here explaining what you're doing? It's clear to me while reviewing the PR but it might be puzzling to somebody that comes along later without that context.